Plugins/Public/mobiledocking_plugin/Main.cpp

```diff
@@ -15,21 +15,57 @@ vector<uint> dockingModuleEquipmentIds;
 // Above how much cargo capacity, should a ship be rejected as a docking user?
 int cargoCapacityLimit = 275;
 
+// How much time will player be given before kick if carrier wants to jettison him.
+int jettisonKickTime = 15;
+
 uint connSystemID = CreateID("li06");
 
-///////////////////////////////////////////////////////////////////////////////////////////////////////////////
-// Function handling changing systems when undocking from a carrier
-void JumpToLocation(uint client, uint system, Vector pos, Matrix ornt)
+// Delayed actions, which need to be done. Look at HkTimerCheckKick().
+vector<ActionJettison> jettisonList;
+
+void HkTimerCheckKick()
+{
+	returncode = DEFAULT_RETURNCODE;
+
+	for (vector<ActionJettison>::iterator it = jettisonList.begin(); it != jettisonList.end(); it++)
+	{
+		it->timeLeft--;
+		if (it->timeLeft == 0)
+		{
+			uint checkCarrierClientID = HkGetClientIdFromCharname(it->carrierCharname);
+			uint checkDockedClientID = HkGetClientIdFromCharname(it->dockedCharname);
+
+			// If both carrier and docked ship are still at server.
+			if (checkDockedClientID != -1 && checkCarrierClientID != -1)
+			{
+				if (mobiledockClients[checkDockedClientID].wscDockedWithCharname == it->carrierCharname)
+				{
+					mobiledockClients[checkCarrierClientID].mapDockedShips.erase(it->dockedCharname);
+					HkKick(HkGetAccountByClientID(checkDockedClientID));

```

Comment:
HkKickReason with ARG_CLIENTID

Reply:
It is HkKick() and it works fine without ARG_CLIENTID

Reply:
Yes but we should give a reason when kicking

Reply:
Now it shows reason.